### PR TITLE
Dev/code cleanup again

### PR DESCRIPTION
**Pull request by NathanielDelgado**
Lets do this again

### CODE REVIEW

**Comment by NathanielDelgado** [comment]
Sorry guys. Merge got screwed so did changes in another branch based on master

**Comment by IshDeshpa** [comment]
🤣🤣🤣

File: Drivers/Src/CanBus.c [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+ */
+void CanBusTxHandler(Can bus) {
+    OS_ERR err = 0;
+    OSSemPost(&(can_mail_sem4[bus]), OS_OPT_POST_1, &err);
+    ASSERT_OS_ERROR(err);
+}
+
+// wrapper functions for the interrupt customized for each bus
+void CanBusTxHandler1() { CanBusTxHandler(kCan1); }
+
```
Comment (diyarajon) [inline Drivers/Src/CanBus.c:50]: Can you remove this blank line if possible/needed to be removed

File: Drivers/Src/CanBus.c [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    }  // if they passed in an invalid id, it will be zero
+
+    // search LUT for id to populate idx and trim data
+    if (entry.idx_en == true) {
+        msg_container->idx = msg_container->data[0];
+        memmove(  // Can't use memcpy, as memory regions overlap
+            msg_container->data, &(msg_container->data[1]),
+            BSP_CAN_DATA_LENGTH -
+                1  // max size of data (8) - size of idx byte (1)
+        );
```
Comment (diyarajon) [inline Drivers/Src/CanBus.c:237]: Would this need to be no linted?

File: Drivers/Src/CanBus.c [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+        &(can_mail_sem4[bus]),
+        (bus == kCan1 ? "CAN Mailbox Semaphore 1" : "CAN Mailbox Semaphore 3"),
+        BSP_CAN_NUM_MAILBOX, &err);  // there's 3 hardware mailboxes on the
+                                     // board, so 3 software mailboxes
+    ASSERT_OS_ERROR(err);
+
+    OSSemCreate(&(can_bus_receive_sem4[bus]),
+                (bus == kCan1 ? "CAN Received Msg Queue Ctr 1"
+                              : "CAN Received Msg Queue Ctr 3"),
+                0, &err);  // create a mailbox counter to hold the messages in
```
Comment (diyarajon) [inline Drivers/Src/CanBus.c:105]: Does this need to be no linted? It's pretty readable anyways

**Review by diyarajon — CHANGES_REQUESTED** [review]
Looks good

File: Drivers/Src/CanBus.c (hunk 1 above)
Comment (NathanielDelgado) [reply Drivers/Src/CanBus.c:50]: Yes

File: Apps/Src/main.c [hunk 4]
```diff
@@ -1,169 +1,127 @@
...
-        (OS_ERR*)&err
-    );
-    assertOSError(err);
+    OSTaskCreate((OS_TCB *)&read_tritium_tcb, (CPU_CHAR *)"ReadTritium",
+                 (OS_TASK_PTR)TaskReadTritium, (void *)NULL,
+                 (OS_PRIO)TASK_READ_TRITIUM_PRIO, (CPU_STK *)read_tritium_stk,
+                 (CPU_STK_SIZE)WATERMARK_STACK_LIMIT,
+                 (CPU_STK_SIZE)TASK_READ_TRITIUM_STACK_SIZE, (OS_MSG_QTY)0,
+                 (OS_TICK)0, (void *)NULL,
+                 (OS_OPT)(OS_OPT_TASK_STK_CLR | OS_OPT_TASK_SAVE_FP),
```
[the remark this answers is not in the document]
Comment (NathanielDelgado) [reply Apps/Src/main.c:78]: Changed

File: BSP/STM32F413/Src/BSP_CAN.c [hunk 5]
```diff
@@ -76,141 +80,160 @@ void BSP_CAN1_Init(uint16_t* idWhitelist, uint8_t idWhitelistSize) {
...
-    CAN_InitStruct.CAN_TXFP = ENABLE;
-    #ifdef CAR_LOOPBACK
+    can_init_struct.CAN_TTCM = DISABLE;
+    can_init_struct.CAN_ABOM = DISABLE;
+    can_init_struct.CAN_AWUM = DISABLE;
+    can_init_struct.CAN_NART = DISABLE;
+    can_init_struct.CAN_RFLM = DISABLE;
+    can_init_struct.CAN_TXFP = ENABLE;
+#ifdef CAR_LOOPBACK
     CAN_InitStruct.CAN_Mode = CAN_Mode_LoopBack;
```
Comment (KnockbackNemo) [inline BSP/STM32F413/Src/BSP_CAN.c:103]:
```suggestion
    can_init_struct.CAN_Mode = CAN_Mode_LoopBack;
```

File: BSP/STM32F413/Src/BSP_CAN.c [hunk 6]
```diff
@@ -220,137 +243,151 @@ void BSP_CAN3_Init(uint16_t* idWhitelist, uint8_t idWhitelistSize)
...
-    CAN_InitStruct.CAN_TXFP = ENABLE;
-    #ifdef MOTOR_LOOPBACK
+    can_init_struct.CAN_TTCM = DISABLE;
+    can_init_struct.CAN_ABOM = DISABLE;
+    can_init_struct.CAN_AWUM = DISABLE;
+    can_init_struct.CAN_NART = DISABLE;
+    can_init_struct.CAN_RFLM = DISABLE;
+    can_init_struct.CAN_TXFP = ENABLE;
+#ifdef MOTOR_LOOPBACK
     CAN_InitStruct.CAN_Mode = CAN_Mode_LoopBack;
```
Comment (KnockbackNemo) [inline BSP/STM32F413/Src/BSP_CAN.c:266]:
```suggestion
    can_init_struct.CAN_Mode = CAN_Mode_LoopBack;
```

**Review by KnockbackNemo — APPROVED** [review]
Looks good to me

**Review by diyarajon — APPROVED** [review]
Looks good!

File: Drivers/Src/CanBus.c (hunk 3 above)
Comment (NathanielDelgado) [reply Drivers/Src/CanBus.c:105]: This is fine I think because the conditional is being another layer so the indent is there

**Pull request merged**